stan/math/rev/mat/functor/integrate_dae.hpp

```diff
@@ -0,0 +1,59 @@
+#ifndef STAN_MATH_REV_MAT_FUNCTOR_INTEGRATOR_DAE_HPP
+#define STAN_MATH_REV_MAT_FUNCTOR_INTEGRATOR_DAE_HPP
+
+#include <stan/math/rev/scal/meta/is_var.hpp>
+#include <stan/math/prim/scal/meta/return_type.hpp>
+#include <stan/math/rev/mat/functor/idas_forward_system.hpp>
+#include <stan/math/rev/mat/functor/idas_integrator.hpp>
+#include <stan/math/prim/scal/err/check_less.hpp>
+#include <algorithm>
+#include <ostream>
+#include <vector>
+
+namespace stan {
+namespace math {
+/**
+ * Return the solutions for a semi-explicit DAE system with residual
+ * specified by functor F,
```

Comment:
I think the functor F should be documented at least somewhere, and I think this is the right place. This is what we did for integrate_1d https://github.com/stan-dev/math/blob/b5a1873eae8fe24a999c84f774aed99d80934b09/stan/math/rev/arr/functor/integrate_1d.hpp#L65 (rev) and https://github.com/stan-dev/math/blob/b5a1873eae8fe24a999c84f774aed99d80934b09/stan/math/prim/arr/functor/integrate_1d.hpp#L152 (prim)